tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/MsTestV2/TestMethodAttributeExecuteIntegration.cs

```diff
@@ -76,92 +79,255 @@ public static class TestMethodAttributeExecuteIntegration
-                            else
-                            {
-                                test.SetErrorInfo(testExceptionType.ToString(), testException.Message, testException.ToString());
-                            }
+                            testTags.EarlyFlakeDetectionTestAbortReason = "slow";
                         }
+                    }
+
+                    var result = HandleTestResult(test, testMethod, returnValueList[i], exception);
+                    allowRetries = allowRetries || result != TestStatus.Skip;
+                }
+            }
+            else
+            {
+                Common.Log.Warning("Failed to extract TestResult from return value");
+                testMethodState.Test.Close(TestStatus.Fail);
+                return new CallTargetReturn<TReturn>(returnValue);
+            }
+
+            if (isTestNew && allowRetries)
+            {
+                // Get retries number
+                int remainingRetries;
+                var slowRetriesSettings = CIVisibility.EarlyFlakeDetectionSettings.SlowTestRetries;
+                if (slowRetriesSettings.FiveSeconds.HasValue && duration.TotalSeconds < 5)
+                {
+                    remainingRetries = slowRetriesSettings.FiveSeconds.Value;
+                    Common.Log.Information<int>("EFD: Number of executions has been set to {Value} for this test that runs under 5 seconds.", remainingRetries);
+                }
+                else if (slowRetriesSettings.TenSeconds.HasValue && duration.TotalSeconds < 10)
+                {
+                    remainingRetries = slowRetriesSettings.TenSeconds.Value;
+                    Common.Log.Information<int>("EFD: Number of executions has been set to {Value} for this test that runs under 10 seconds.", remainingRetries);
+                }
+                else if (slowRetriesSettings.ThirtySeconds.HasValue && duration.TotalSeconds < 30)
+                {
+                    remainingRetries = slowRetriesSettings.ThirtySeconds.Value;
+                    Common.Log.Information<int>("EFD: Number of executions has been set to {Value} for this test that runs under 30 seconds.", remainingRetries);
+                }
+                else if (slowRetriesSettings.FiveMinutes.HasValue && duration.TotalMinutes < 5)
+                {
+                    remainingRetries = slowRetriesSettings.FiveMinutes.Value;
+                    Common.Log.Information<int>("EFD: Number of executions has been set to {Value} for this test that runs under 5 minutes.", remainingRetries);
+                }
+                else
+                {
+                    remainingRetries = 1;
+                    Common.Log.Information("EFD: Number of executions has been set to 1. Current test duration is {Value}", duration);
+                }
 
-                        if (!string.IsNullOrEmpty(testResult.DisplayName) && test.Name != testResult.DisplayName)
+                if (--remainingRetries > 0)
```

Comment:
🤨 what do you have against --[x]?

Reply:
changed in [b12fb8e](https://github.com/DataDog/dd-trace-dotnet/pull/5320/commits/b12fb8e0c416c402a0bf4747deac0d9b4a9709ac)

Reply:
Me, not a lot, but it's one of those things that I often see quoted as unnecessary complexity, because it's something else you have to understand `x--` vs `--x` 🤷‍♂️

Reply:
those people never saw a C++ compiler error for a missing ';' or what? :old_man_yells:

Reply:
😆 Correct, they have not 😛